src/box/lua/connpool.lua

```diff
@@ -348,9 +429,12 @@ local function get_connection(opts)
 
     -- Increase weight of local candidate.
     if opts.prefer_local ~= false then
+        local local_instance_name = box.info.name
+
         local weight_local = 1
-        if weights[box.info.name] ~= nil then
-            weights[box.info.name] = weights[box.info.name] + weight_local
+        if weights[local_instance_name] ~= nil then
+            weights[local_instance_name] = weights[local_instance_name] +
+                                           weight_local
```

Comment:
Please, don't complicate understanding of a logic behind your patches by such irrelevant changes. At least move them to its own commits.

This particular name now can be replaced to `config:info('v2').hierarchy.instance` (in a separate PR). So, I wouldn't touch it.